File: lib/models/mistake_pattern.dart
class MistakePattern {
  final String id;
  final String category;
  final String label;
  final String description;
  final String userExample;
  final String betterJapanese;
  final String betterPronunciation;
  final String betterTranslation;
  final int count;
  final String lastTopic;

  MistakePattern({
    required this.id,
    required this.category,
    required this.label,
    required this.description,
    required this.userExample,
    required this.betterJapanese,
    required this.betterPronunciation,
    required this.betterTranslation,
    required this.count,
    required this.lastTopic,
  });

  factory MistakePattern.fromJson(Map<String, dynamic> json) {
    return MistakePattern(
      id: json['id'] as String? ?? '',
      category: json['category'] as String? ?? '표현',
      label: json['label'] as String? ?? '',
      description: json['description'] as String? ?? '',
      userExample: json['userExample'] as String? ?? '',
      betterJapanese: json['betterJapanese'] as String? ?? '',
      betterPronunciation: json['betterPronunciation'] as String? ?? '',
      betterTranslation: json['betterTranslation'] as String? ?? '',
      count: json['count'] as int? ?? 1,
      lastTopic: json['lastTopic'] as String? ?? '',
    );
  }
}

File: backend/app/models/mistake_pattern.py
from pydantic import BaseModel, Field


class MistakePatternResponse(BaseModel):
    id: str
    category: str
    label: str
    description: str = ""
    user_example: str = Field(default="", alias="userExample")
    better_japanese: str = Field(default="", alias="betterJapanese")
    better_pronunciation: str = Field(default="", alias="betterPronunciation")
    better_translation: str = Field(default="", alias="betterTranslation")
    count: int = 1
    last_topic: str = Field(default="", alias="lastTopic")
    updated_at: str = Field(default="", alias="updatedAt")

    model_config = {"populate_by_name": True, "by_alias": True}


class MistakePatternListResponse(BaseModel):
    patterns: list[MistakePatternResponse]
